paypal/paypal-ios
# Add `AnalyticsService` and use single `sessionID`

### Reason for changes

- Re-write of #100 
    - Punting 🏈 on [the singleton discussion](https://github.com/paypal/iOS-SDK/pull/100#discussion_r1042254899) until we need to handle batching
- Follow up PR to #91

### Summary of changes

- Add `AnalyticsService`
    - Manages a static `sessionID`, constructs the appropriate `AnalyticsEventData`, and calls the `HTTP` class to perform the actual request.
    - This lets us move our Analytics specific logic outside of `APIClient` & `HTTP`.
- Add `AnalyticsService_Test`
    - Confirm single `sessionID` used across instances
    - Add `MockHTTP` to confirm `AnalyticsEventRequest` type sent

### Checklist

- ~Added a changelog entry~

### Authors
@scannillo
